app/controllers/articles_controller.rb

```diff
@@ -35,4 +36,16 @@ def set_categories
       active_category category.parent_id if category.child?
     end
   end
+
+  def set_show_newsletter_signup
+    exclusions = YAML.load(
+      File.read(File.expand_path("#{Rails.root}/config/newsletter_article_exclusion_list.yml", __FILE__))
```

Comment:
Another thing just for showing an improvement. You can make the same thing with:

```ruby
 File.read(Rails.root.join('config', 'newsletter_article_exclusion_list.yml'))
```